asset-transfer-basic/application-gateway-java/src/main/java/application/java/App.java

```diff
@@ -0,0 +1,251 @@
+
+        System.out.println("Submitting \"put\" for readAssetById transaction with arguments: time, " + timestamp);
+
+        // Submit a transaction, blocking until the transaction has been committed on the ledger.
+        byte[] submitResult = contract.submitTransaction("ReadAsset", assetId, "time", timestamp);
+
+        System.out.println("Submit result: " + new String(submitResult, StandardCharsets.UTF_8));
+        System.out.println("Evaluating \"get\" get query for readAssetById with arguments: time");
+
+        byte[] evaluateResult = contract.evaluateTransaction("ReadAsset", "time");
+        System.out.println("Query result: " + new String(evaluateResult, StandardCharsets.UTF_8));
+    }
+    
+    private static void updateNonExistentAsset(Contract contract) throws GatewayException, CommitException {
+
+        String timestamp = LocalDateTime.now().toString();
+        System.out.println("Submitting \"put\" transaction for updateNonExistentAsset with arguments: time, " + timestamp);
+
+        // Submit a transaction, blocking until the transaction has been committed on the ledger.
+        byte[] submitResult = contract.submitTransaction("UpdateAsset","asset70","blue","5","Tomoko","300");
+
+        System.out.println("Submit result: " + new String(submitResult, StandardCharsets.UTF_8));
+        System.out.println("Evaluating \"get\" query for updateNonExistentAsset with arguments: time");
+
+        byte[] evaluateResult = contract.evaluateTransaction("UpdateAsset", "time");
+        System.out.println("Query result: " + new String(evaluateResult, StandardCharsets.UTF_8));
+    }
+    
+    
+    private static ManagedChannel newGrpcConnection() throws IOException, CertificateException {
+        Reader tlsCertReader = Files.newBufferedReader(tlsCertPath);
+        X509Certificate tlsCert = Identities.readX509Certificate(tlsCertReader);
+
+        return NettyChannelBuilder.forTarget("localhost:7051")
+                .sslContext(GrpcSslContexts.forClient().trustManager(tlsCert).build())
+                .overrideAuthority("peer0.org1.example.com")
+                .build();
+    }
+
+    private static Identity newIdentity() throws IOException, CertificateException {
+        Reader certReader = Files.newBufferedReader(certPath);
+        X509Certificate certificate = Identities.readX509Certificate(certReader);
+
+        return new X509Identity(mspID, certificate);
+    }
+
+    private static Signer newSigner() throws IOException, InvalidKeyException {
+        File dir = new File(keyPath.toString());
+        File[] listOfFiles = dir.listFiles();
+        //System.out.println("\n" + "The filename =" + listOfFiles[0] + "\n type = " + (listOfFiles[0]).getClass());
+        Path path = Paths.get(listOfFiles[0].getPath());
+        Reader keyReader = Files.newBufferedReader(path);
+        PrivateKey privateKey = Identities.readPrivateKey(keyReader);
+
+        return Signers.newPrivateKeySigner(privateKey);
+    }
```

Comment:
Move these up in the file so the methods appear after `main()` in the same order in which `main()` invokes them.

Reply:
ok